### PR TITLE
Fix for Issue 2031: Updated Admin/Grader Spreadsheet marks table to React

### PR DESCRIPTION
**Pull request by peterzhao333**
Updated the marks spreadsheet table for the Admin/Grader's view to React

### CODE REVIEW

File: app/controllers/grade_entry_forms_controller.rb [hunk 1]
```diff
@@ -270,10 +267,55 @@ def get_mark_columns
...
+      {
+        id: 'marking_state',
+        content: t('grade_entry_forms.grades.marking_state')
+      }
+    end
 
     render json: c
   end
 
+  def populate_grades_table
```
Comment (peterzhao333) [inline app/controllers/grade_entry_forms_controller.rb]: Not sure what this means?

Comment (arkon) [inline app/controllers/grade_entry_forms_controller.rb]: https://github.com/bbatsov/rubocop/blob/master/lib/rubocop/cop/metrics/abc_size.rb

File: app/controllers/grade_entry_forms_controller.rb [hunk 2]
```diff
@@ -230,9 +230,8 @@ def update_grade
     @grade_entry_item_id = params[:grade_entry_item_id]
     updated_grade = params[:updated_grade]
 
-    grade_entry_student =
-      grade_entry_form.grade_entry_students.find_or_create_by_user_id(
-            @student_id)
+    grade_entry_student = grade_entry_form.grade_entry_students
+                          .find_or_create_by_user_id(@student_id)
```
Comment (david-yz-liu) [inline app/controllers/grade_entry_forms_controller.rb]: Align the two periods

File: app/controllers/grade_entry_forms_controller.rb [hunk 3]
```diff
@@ -253,27 +252,69 @@ def student_interface
...
       c <<
-      {
-        id: 'total_marks',
-        content: t('grade_entry_forms.grades.total') \
-                 + ' ' + grade_entry_form.out_of_total.to_s,
-      }
+        {
+          id: 'total_marks',
+          content: t('grade_entry_forms.grades.total') \
+                   + ' ' + grade_entry_form.out_of_total.to_s,
```
Comment (david-yz-liu) [inline app/controllers/grade_entry_forms_controller.rb]: Can you change this to put the total in parentheses as well, just like the other columns?

File: app/controllers/grade_entry_forms_controller.rb (hunk 1 above)
Comment (david-yz-liu) [inline app/controllers/grade_entry_forms_controller.rb]: Eh, I'm okay ignoring this comment for now. But thanks @arkon!

File: app/controllers/grade_entry_forms_controller.rb [hunk 4]
```diff
@@ -253,27 +252,69 @@ def student_interface
...
 
+  def populate_grades_table
+    @grade_entry_form = GradeEntryForm.find(params[:id])
+    @students = Student.all
+
+    @student_grades = @students.map do |student|
+      s = student.attributes
+      student_grade_entry = @grade_entry_form.grade_entry_students
+                            .find_by_user_id(student.id)
+      if !student_grade_entry.nil?
```
Comment (david-yz-liu) [inline app/controllers/grade_entry_forms_controller.rb]: use `unless`

File: app/controllers/grade_entry_forms_controller.rb [hunk 5]
```diff
@@ -253,27 +252,69 @@ def student_interface
...
   end
 
+  def populate_grades_table
+    @grade_entry_form = GradeEntryForm.find(params[:id])
+    @students = Student.all
+
+    @student_grades = @students.map do |student|
+      s = student.attributes
+      student_grade_entry = @grade_entry_form.grade_entry_students
+                            .find_by_user_id(student.id)
```
Comment (david-yz-liu) [inline app/controllers/grade_entry_forms_controller.rb]: align the periods

File: app/controllers/grade_entry_forms_controller.rb [hunk 6]
```diff
@@ -253,27 +252,69 @@ def student_interface
...
+          content: t('grade_entry_forms.grades.marking_state')
+        }
     end
 
     render json: c
   end
 
+  def populate_grades_table
+    @grade_entry_form = GradeEntryForm.find(params[:id])
+    @students = Student.all
```
Comment (david-yz-liu) [inline app/controllers/grade_entry_forms_controller.rb]: I think you should remove this, and just use it inline with the `map` below

File: app/views/grade_entry_forms/_react_grades_table.js.jsx.erb [hunk 7]
```diff
@@ -0,0 +1,269 @@
...
+  // Prevent the enter key from releasing grades
+    preventEnterSubmit();
+  });
+
+  /** This function will prevent the default HTML5 action of submitting the content form
+    *  when the user presses "enter". As an Admin user, this would release the grades.
+    */
+  function preventEnterSubmit() {
+    jQuery(document).keypress(function(event) {
+      if (event.which == 13) {
```
Comment (david-yz-liu) [inline app/views/grade_entry_forms/_react_grades_table.js.jsx.erb]: Please use ===

File: app/views/grade_entry_forms/_react_grades_table.js.jsx.erb [hunk 8]
```diff
@@ -0,0 +1,269 @@
...
+              type: 'POST',
+              dataType: 'script',
+            });      
+          });
+        });
+      }        
+    },
+
+    refresh: function() {
+     document.getElementById('working').style.display = '';
```
Comment (david-yz-liu) [inline app/views/grade_entry_forms/_react_grades_table.js.jsx.erb]: This needs to be indented by one more space

**Comment by david-yz-liu** [comment]
@peterzhao333 This is looking good. Since the marks can't be entered right now, I'm going to make a separate PR to the "spreadsheets" branch that I've creaed for you.

**Pull request closed without merging**